nsq/reader.py

```diff
@@ -677,7 +669,7 @@ def _redistribute_rdy_state(self):
             # We also don't attempt to avoid the connections who previously might have had RDY 1
             # because it would be overly complicated and not actually worth it (ie. given enough
             # redistribution rounds it doesn't matter).
-            possible_conns = list(self.conns.values())
+            possible_conns = [c for c in conns if not (c.in_flight or c.rdy)]
```

Comment:
OK, I think that makes sense, but what about the section above? The sum of `in_flight` can also be less than `total_rdy`, in which case we'd enter this loop below thinking we have more than we actually do to give out?